contracts/cw-blotto/src/contract.rs

```diff
@@ -435,25 +470,31 @@ impl BlottoContract<'_> {
             }
         }
 
-        // Load game winner and prize pool
-        let game_winner = self.winner.load(ctx.deps.storage)?;
-        let prize_pool = self.prize_pool.load(ctx.deps.storage)?;
-
-        // Load the total amount the player staked to the winning army
-        let players_stake = self
-            .player_totals_by_army
-            .may_load(ctx.deps.storage, (&ctx.info.sender, game_winner.id))?;
-
-        if let Some(players_stake) = players_stake {
-            // TODO this math may have edge cases?
-            // Calculate players share of the prize pool
-            // players stake * prize pool / total staked on winning army
-            let winnings = players_stake
-                .checked_mul(prize_pool)?
-                .checked_div(game_winner.total_staked)?;
-
-            // Add player's share of the prize pool to the withdraw_amount
-            withdraw_amount = withdraw_amount.checked_add(winnings)?;
+        // Load game winner
+        // If a game ended with a tie there is no winner entry stored
+        let game_winner = self.winner.may_load(ctx.deps.storage)?;
+        if let Some(game_winner) = game_winner {
+            // Load the prize pool
+            let prize_pool = self.prize_pool.load(ctx.deps.storage)?;
+
+            // Load the total amount the player staked to the winning army
+            let players_stake = self
+                .player_totals_by_army
+                .may_load(ctx.deps.storage, (&ctx.info.sender, game_winner.id))?;
+
+            println!("players_stake: {:?}", players_stake);
+
+            if let Some(players_stake) = players_stake {
+                // TODO this math may have edge cases?
+                // Calculate players share of the prize pool
+                // players stake * prize pool / total staked on winning army
+                let winnings = players_stake
+                    .checked_mul(prize_pool)?
+                    .checked_div(game_winner.total_staked)?;
+
+                // Add player's share of the prize pool to the withdraw_amount
+                withdraw_amount = withdraw_amount.checked_add(winnings)?;
+            }
```

Comment:
We need to return user funds if there is a tie! Would use a `match` statement here, and if there is no game winner get the player's total stake and they can withdraw everything.